distributed/worker.py

```diff
@@ -1563,12 +1566,7 @@ def put_key_in_memory(self, key, value, transition=True):
         if stop - start > 0.020:
             self.startstops[key].append(('disk-write', start, stop))
 
-        if key not in self.nbytes:
-            self.nbytes[key] = sizeof(value)
-
-        if key not in self.types:
-            self.types[key] = type(value)
-
+        self.nbytes[key] = sizeof(value)
```

Comment:
We assume immutability.  Additionally sizeof can take some time while type is free.